editor/components/default-block-appender/index.js

```diff
@@ -33,7 +40,16 @@ export function DefaultBlockAppender( { onAppend, showPrompt = true } ) {
 }
 
 export default connect(
-	null,
+	( state, ownProps ) => {
+		const isEmpty = ! getBlockCount( state, ownProps.rootUID );
```

Comment:
Nice refactoring. It is much cleaner when this component decides when to render.